### DISCUSSION TITLE
Problems with version changes.

### DISCUSSION THREAD

**Discussion by lpalacecr** [primefaces/community#1321] [PrimeVue] [answered]
Problems with version changes.

From version 3.47.0 to the current 3.49.1, paid themes and even primeblock do not work correctly. Very important corrections have been made to the components and new ones have also been added. I have no problem with the new components, I just don't use them, but I need the existing components, and I need them as stable as possible. According to NPM versioning theory, we have a.b.c where b and c should maintain compatibility with previous versions, b for minor changes and c for patches. In version 3.47.2 they included a change that is not compatible with previous versions, which has meant that for those of us who bought the themes, we cannot continue upgrading the version and remain stuck at version 3.46.0; According to theory, version 3.47.2 had to have been version 4.0.0, with this it is understood that you will see incompatibilities. Every time they update the version I see that they combine improvements, patches and new components (when there are any), which leaves us without the possibility of being able to update the patches. It seems essential to me that the patches are managed independently of the improvements and new components and that the changes that cause incompatibilities are uploaded to the version, in order to have access to patch updates.
Primevue, although it is true, is open source, the themes are not, which is why the priority should change to the paid ones as long as they continue with the current update scheme. I have a bank credit system, it is a very complex and delicate system due to the large volumes of money it controls, that is why I need the components to be 100%, in short, I need to update all the patches.

PSI: @cagataycivici @tugcekucukoglu

Translated by Google Translate.

Original text:

Problemas con los cambios de versión.

Desde la versión 3.47.0 hasta la actual 3.49.1 los temas de pago e inclusive el primeblock no funcionan correctamente. Se han hecho correciones muy importantes en los componentes y también se han agreado nuevos. Con los componentes nuevos no tengo problema, simplemente no los uso, pero los componentes ya existentes los necesito, y los necesito lo más estable posible. Según dicta la teoría de versionamiento del NPM tenemos a.b.c donde b y c deberían mantener compatibilidad con las versiones anteriores, b para cambios menores y c para parches. En la versión 3.47.2 incluyeron un cambio no compatible con las versiones anteriores lo que ha provocado que para los que compramos los temas, no podamos seguir subiendo de versión y quedarnos estancados en la versión 3.46.0; según la teoría la versión 3.47.2 tuvo que haber sido la versión 4.0.0, con eso se sobreentiende que van a ver incompatibilidades. Cada vez que actualizan la version veo que combinan mejoras, parches y componentes nuevos (cuando los hay), lo que nos deja sin la posibilidad de poder actualizar los parches. Me parece fundamental que los parches se manejen independientemente de las mejoras y componentes nuevos y que los cambios que provoquen incompatibilidades subirlos de versión, para así tener acceso a la actualización de parches.
El primevue sin bien es cierto es open source, los temas no, motivo por el cual la prioridad debe cambiar a los que son de pago mientras sigan con el esquema de actualizaciones actual. Yo tengo un sistema de créditos bancarios, es un sistema muy complejo y delicado por los grandes volumenes de dinero que controla, por eso necesito que los componentes esten al 100%, en resumen, necesito actualizar todos los parches.

PSI: @cagataycivici @tugcekucukoglu

**Comment by tugcekucukoglu** [comment, marked as answer]
The new version of templates is released on this Monday and we're currently working on the PrimeBlocks. We're moving a new design system and that creates some changes. 

Thanks for your understanding.
